### PR TITLE
[NEW] Add  subscriptions.getAll and test

### PR DESCRIPTION
**Pull request by ulietaight**
'subscriptions.getAll' - rest api method added
test for 'subscriptions.getAll'  added

### CODE REVIEW

**Comment by pafik13** [comment]
@ulietaight I tried it - doesn't work :(  Response is empty always....

**Comment by ulietaight** [comment]
> @ulietaight I tried it - doesn't work :( Response is empty always....

There is a check for the current user's ID. Perhaps the request was executed without a login.

**Comment by pafik13** [comment]
> > @ulietaight I tried it - doesn't work :( Response is empty always....
> 
> There is a check for the current user's ID. Perhaps the request was executed without a login.

Here is an example of my call:

```
curl -H "X-Auth-Token: X1GgL-vNlKWeKzYxA0Yd2y96Z5koEMMRGzvOEpfvRNc" \
     -H "X-User-Id: default.admin" \
     'http://localhost:3000/api/v1/subscriptions.getAll'
```

**Comment by ulietaight** [comment]
> > > @ulietaight I tried it - doesn't work :( Response is empty always....
> > 
> > 
> > There is a check for the current user's ID. Perhaps the request was executed without a login.
> 
> Here is an example of my call:
> 
> ```
> curl -H "X-Auth-Token: X1GgL-vNlKWeKzYxA0Yd2y96Z5koEMMRGzvOEpfvRNc" \
>      -H "X-User-Id: default.admin" \
>      'http://localhost:3000/api/v1/subscriptions.getAll'
> ```

try it now by specifying userId in queryParams

**Comment by pafik13** [comment]
> > > > @ulietaight I tried it - doesn't work :( Response is empty always....
> > > 
> > > 
> > > There is a check for the current user's ID. Perhaps the request was executed without a login.
> > 
> > 
> > Here is an example of my call:
> > ```
> > curl -H "X-Auth-Token: X1GgL-vNlKWeKzYxA0Yd2y96Z5koEMMRGzvOEpfvRNc" \
> >      -H "X-User-Id: default.admin" \
> >      'http://localhost:3000/api/v1/subscriptions.getAll'
> > ```
> 
> try it now by specifying userId in queryParams

Ok. Now it works...

**Comment by pafik13** [comment]
@ulietaight what's next:

1. rename to `meteor-subscriptions`
2. change `authRequired` to `true` (for security reason)
3. analyze code in `notifications` for `streamRoom` and try to extract additional subscription:
 
 * https://github.com/pafik13/Rocket.Chat/blob/develop/packages/rocketchat-notifications/server/lib/Notifications.js#L96
 * https://github.com/pafik13/Rocket.Chat/blob/develop/packages/rocketchat-notifications/server/lib/Notifications.js#L73

File: tests/end-to-end/api/10-subscriptions.js [hunk 1]
```diff
@@ -200,4 +200,17 @@ describe('[Subscriptions]', function() {
 				.end(done);
 		});
 	});
+	describe('[/subscriptions.getAll]', () => {
```
Comment (pafik13) [inline tests/end-to-end/api/10-subscriptions.js]: @ulietaight тест теперь кривой и не на своём месте - надо перенести в `misc`

**Review by pafik13 — APPROVED** [review]
LGFM

**Pull request merged**